examples/camera/2d_top_down_camera.rs

```diff
@@ -0,0 +1,137 @@
+        level: Level::INFO,
+        ..default()
+    });
+
+    App::new()
+        .add_plugins(default_plugins)
+        .insert_resource(ClearColor(Color::BLACK))
+        .add_systems(Startup, (scene_setup, camera_setup).chain())
+        .add_systems(Update, (move_player, update_camera).chain())
+        .run();
+}
+
+fn scene_setup(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<ColorMaterial>>,
+) {
+    // World where we move the player
+    commands.spawn(MaterialMesh2dBundle {
+        mesh: Mesh2dHandle(meshes.add(Rectangle::new(1000., 700.))),
+        material: materials.add(Color::srgb(52. / 255., 39. / 255., 69. / 255.)), //77 57 102
+        ..default()
+    });
+
+    // Player
+    commands.spawn(PlayerBundle {
+        player: Player,
+        mesh_bundle: MaterialMesh2dBundle {
+            mesh: meshes.add(Circle::new(25.)).into(),
+            material: materials.add(Color::srgb(6.25, 9.4, 9.1)),
+            transform: Transform {
+                translation: vec3(0., 0., 2.),
+                ..default()
+            },
+            ..default()
+        },
+    });
+
+    debug!("Scene setup finished");
+}
+
+fn camera_setup(mut commands: Commands) {
+    let camera = Camera2dBundle {
+        transform: Transform::from_xyz(0., 0., 1.),
+        camera: Camera {
+            hdr: true,
+            ..default()
+        },
+        ..default()
+    };
+
+    commands.spawn((camera, BloomSettings::NATURAL));
+
+    debug!("Camera setup finished");
+}
+
+fn update_camera(
+    mut camera: Query<&mut Transform, (With<Camera2d>, Without<Player>)>,
+    player: Query<&Transform, (With<Player>, Without<Camera2d>)>,
+    time: Res<Time>,
+) {
+    let Ok(mut camera) = camera.get_single_mut() else {
+        debug!("Camera2d not found");
+        return;
+    };
+
+    let Ok(player) = player.get_single() else {
+        debug!("Player not found");
+        return;
+    };
+
+    let Vec3 { x, y, .. } = player.translation;
+    let direction = Vec3::new(x, y, camera.translation.z);
+
+    // Move camera with a smooth effect
+    camera.translation = camera
+        .translation
+        .lerp(direction, time.delta_seconds() * CAM_LERP_FACTOR);
+}
+
+fn move_player(
+    mut player: Query<&mut Transform, With<Player>>,
+    time: Res<Time>,
+    kb_input: Res<ButtonInput<KeyCode>>,
+) {
+    let Ok(mut player) = player.get_single_mut() else {
+        debug!("Player not found");
+        return;
+    };
+
+    let mut direction = vec2(0., 0.);
+
+    if kb_input.pressed(KeyCode::KeyW) {
+        direction.y = 1.;
+    }
+
+    if kb_input.pressed(KeyCode::KeyS) {
+        direction.y = -1.;
+    }
+
+    if kb_input.pressed(KeyCode::KeyA) {
+        direction.x = -1.;
+    }
+
+    if kb_input.pressed(KeyCode::KeyD) {
+        direction.x = 1.;
+    }
+
+    let move_delta = direction * PLAYER_SPEED * time.delta_seconds();
```

Comment:
I think I already normalized the direction with the input handler? `x` and `y` can only be 0.0, -1.0 or 1.0

Reply:
See https://forum.unity.com/threads/diagonal-movement-is-quicker.941102/#:~:text=You%20need%20to%20normalize%20the,construct%20from%20the%20input%20axes.&text=As%20you%20can%20see%2C%20when,in%20terms%20of%20a%20square. for example.

The magnitude of the vector exceeds 1 when moving diagonally. I believe you should be able to observe the before and after: before the example moves faster when going diagonally.

Reply:
Thank you! I didn't notice this before. Really interesting and useful to know! Even contributing to examples can help you to learn a lot! I pushed the fix along with a comment - I thought it would be a great thing to add for guidance.